Tallies/TallyClerks/eventClerk_class.f90

```diff
@@ -0,0 +1,421 @@
+      if (self % filter % isFail(state)) return
+    end if
+
+    ! Find bin index
+    if (allocated(self % map)) then
+      binIdx = self % map % map(state)
+    else
+      binIdx = 1
+    end if
+
+    ! Return if invalid bin index
+    if (binIdx == 0) return
+
+    ! Convert the particle to an event
+    ! Store the event in the thread-private bank
+    id = ompGetThreadNum() + 1
+    bin = self % lastEntry(id) + 1
+    call self % scores(bin, id) % build(p, MT)
+    self % lastEntry(id) = bin
+
+    ! If the bank is full, flush it to output
+    if (bin == self % frequency) then
+      !$omp critical
+      ! Escape in case all events were completed while waiting on
+      ! the critical
+      if (self % totalEvents >= self % maxEvents) then
+        self % lastEntry(id) = 0
+      else
+
+        call self % outputScores(id)
+      
+        ! Increment the total number of events
+        self % totalEvents = self % totalEvents + self % lastEntry(id)
+        self % lastEntry(id) = 0
+      end if
+      !$omp end critical
+    end if
+
+  end subroutine reportOutColl
+
+  !!
+  !! Display convergence progress on the console
+  !!
+  !! See tallyClerk_inter for details
+  !!
+  subroutine display(self, mem)
+    class(eventClerk), intent(in) :: self
+    type(scoreMemory), intent(in) :: mem
+
+    print *, 'eventClerk does not support display yet'
+
+  end subroutine display
+
+  !!
+  !! Write contents of the clerk to output file.
+  !! In this case, only the map is written to the usual output.
+  !! Meanwhile, the events are written to the event file if they
+  !! haven't already been flushed.
+  !!
+  !! See tallyClerk_inter for details
+  !!
+  subroutine print(self, outFile, mem)
+    class(eventClerk), intent(in)              :: self
+    class(outputFile), intent(inout)           :: outFile
+    type(scoreMemory), intent(in)              :: mem
+    integer(shortInt)                          :: i
+
+    ! Make sure that score arrays have been flushed already.
+    ! If not, write the remainder to output.
+    do i = 1, size(self % lastEntry)
+      if (self % lastEntry(i) > 0) call self % outputScores(i)
+    end do
+
+    ! Write out any map that was used to the usual output file,
+    ! but scores are written out separately.
+
+    ! Begin block
+    call outFile % startBlock(self % getName())
+
+    ! If collision clerk has map print map information
+    if (allocated(self % map)) then
+      call self % map % print(outFile)
+    end if
+    
+    call outFile % endBlock()
+
+  end subroutine print
+
+  !!
+  !! Send events in a thread's bank to the output file
+  !!
+  subroutine outputScores(self, id)
+    class(eventClerk), intent(in) :: self
+    integer(shortInt), intent(in) :: id
+    integer(shortInt)             :: i, u, ios
+
+    associate(ev => self % scores(:,id))
+
+
+      ! Open the file and append the event info
+      open(newunit=u, file=trim(self % outputFile), status='unknown', &
```

Comment:
Here you are either creating the file or opening it, right? Is `append` happy with either?